### ISSUE TITLE
24h format

### ISSUE THREAD

**Issue by sevanteri** [groyoh/minimalin#76]
Hey,

Just creating this issue before creating a pull request, like you asked.

The Pebble C API already includes a function ([`clock_is_24h_style`](https://developer.pebble.com/docs/c/Foundation/Wall_Time/#clock_is_24h_style)) for checking the user preferences for using the 24h or the 12h format. I have included this in the `update_current_time` function as follows:

```
  if (!clock_is_24h_style()) {
    if(hour > 12){
      hour -= 12;
    }else if(hour == 0){
      hour = 12;
    }
  }
```

So the formatting to 12h format is skipped like the user prefers.

I'm also curious why [`strftime`](https://developer.pebble.com/docs/c/Standard_C/Time/#strftime) isn't used. As there already is a function for formatting the dates and times, it puzzles me why you are doing the formatting yourself. Using `strftime` would simplify the code a bit in my opinion.

**Comment by AlessioLaiso** [comment]
Hi Pete, thanks for creating the issue!

We would like to add the option for 24h format in the configuration. We do not want the format to be set automatically to 24h if the phone is set to 24h. This is because Minimalin is still (sort of) an analog watchface, and we figured that many users would like to keep the analog analogy (pun not intended) using a 12h format. I hope it’s clear.

TL;DR: 24h is cool, but we think it should be an opt-in setting in the configuration page.

**Comment by sevanteri** [comment]
This setting is set in the Pebble watch's own settings and this modification doesn't affect the clock hands in any way, just the numbers.

**Comment by groyoh** [comment]
@sevanteri there was a specific reason for not using `strftime` which I don't recall at the moment. And i don't think you got @AlessioLaiso's point. We want to add 24-hour option BUT we want to add it as a configuration and not use the user's default style. There are already some work going on for this, I'll try to put it out tonight if possible.

**Comment by sevanteri** [comment]
Yup, misunderstood earlier.

And now that I do understand, I kinda like the way it is now. :)

**Comment by groyoh** [comment]
@sevanteri just so you know. We WILL provide this feature ;) 
![pebble_screenshot_2016-05-03_18-51-50](https://cloud.githubusercontent.com/assets/7434196/14991121/1d40eb4e-1160-11e6-8b74-aa712fc6153a.png)

**Comment by groyoh** [comment]
@sevanteri it's released in the 1.3 version ;)
